packages/hypixel-api-client/src/index.ts

```diff
@@ -0,0 +1,183 @@
+/**
+ * Copyright (c) Statsify
+ *
+ * This source code is licensed under the GNU GPL v3 license found in the
+ * LICENSE file in the root directory of this source tree.
+ * https://github.com/Statsify/statsify/blob/main/LICENSE
+ */
+
+import Axios, { AxiosError, AxiosInstance } from "axios";
+import { HypixelPlayer } from "./types";
+import { setTimeout } from "node:timers/promises";
+
+export class HypixelKeyRejectedError extends Error {
+  public constructor(message: string) {
+    super(message);
+    this.name = "EHYAPIKEY";
+  }
+}
+
+export class HypixelAPIError extends Error {
+  public constructor(message: string) {
+    super(message);
+    this.name = "EHYAPI";
+  }
+}
+
+export type HypixelAPIConfig = {
+  key: string;
+  autoSleep: boolean;
+};
+
+export class HypixelAPI {
+  public limit: number;
+  public remaining: number;
+
+  private axios: AxiosInstance;
+  private config: HypixelAPIConfig;
+
+  public constructor(config: HypixelAPIConfig) {
+    this.axios = Axios.create({
+      baseURL: "https://api.hypixel.net/",
+      headers: { "API-KEY": config.key },
+    });
+    this.config = config;
+  }
+
+  public player(uuid: string): Promise<HypixelPlayer | undefined> {
+    return this.get<HypixelPlayer>("player", { uuid });
+  }
+
+  public friends(uuid: string) {
+    return this.get("friends", { uuid });
+  }
+
+  public recentgames(uuid: string) {
+    return this.get("recentgames", { uuid });
+  }
+
+  public status(uuid: string) {
+    return this.get("status", { uuid });
+  }
+
+  public guild(params: { id?: string; player?: string; name?: string }) {
```

Comment:
Make a type called `GuildParams`
```ts
type GuildParams = { id: string } | { uuid: string } | { name: string }
```

```suggestion
  public guild(params: GuildParams) {
```